### DISCUSSION TITLE
WatchSystemPreference

### DISCUSSION THREAD

**Discussion by henriquebg88** [balta-io/forum#714] [Tira Dúvidas] [answered]
Tem algo errado para o meu não funcionar a alteração quando mudo no sistema? 
Pelo clique no Switch funciona. 

Estou em um Windows 11 Home .

![image](https://github.com/user-attachments/assets/5b5f89ae-3265-4248-8ed8-331243c2e794)
`@code{
    private bool isDarkMode = true;
    private MudThemeProvider mudThemeProvider = null!;

    // Depois que todos os elementos foram renderizados na tela
    protected override async Task OnAfterRenderAsync(bool firstRender){
        if(firstRender){
            isDarkMode = await mudThemeProvider.GetSystemPreference();

            await mudThemeProvider.WatchSystemPreference(OnSystemThemeChange);

            StateHasChanged();
        }

    }

    private Task OnSystemThemeChange(bool newValue){
        isDarkMode = newValue;
        StateHasChanged(); // Para alguns tipos de ocasiões, o Blazor não irá atualizar a tela automáticamente
        return Task.CompletedTask;
    }
}`

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @henriquebg88, provavelmente... copia e cola o código do curso e veja se funciona...
Se funcionar, foi algo que digitou errado!

**Comment by ronrodrigues** [comment]
Boa noite.
Estou tendo o mesmo problema que o @henriquebg88.
Copiei o código do MainLayout.razor do curso, mas ainda assim não funciona.
Quando troco o tema, não é identificado.
Debugando, encontrei o motivo: GetSystemPreference() sempre está retornando falso, mesmo estando com tema dark.

No meu caso, uso Linux (Fedora 42 Gnome), mas pode ser que seja por conta da versão do .NET (9.0.106) ou do MudBlazor (8.6.0).

![image](https://github.com/user-attachments/assets/248c4c74-ce67-4a81-94d2-cf893c19af41)

**Reply by andrebaltieri** [reply]
**Acho** que no Linux não tem suporte :/

**Reply by ronrodrigues** [reply]
> **Acho** que no Linux não tem suporte :/

Que pena. Mas, tudo bem então. Obrigado pelo retorno, Balta.

**Reply by andrebaltieri** [reply]
Pois é... tenta outro navegador, quem sabe!

**Reply by ronrodrigues** [reply]
> Pois é... tenta outro navegador, quem sabe!

Até tentei. Como utilizo Gnome usei o Gnome Web (Epiphany) mas também não funcionou. O navegador e todo o sistema muda, a página não. =(
